lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -176,13 +176,40 @@ def credit(amount, payment, options = {})
         end
       end
 
-      def verify(credit_card, options = {})
+      def verify(payment_method, options = {})
+        amount = define_amount_for_verify(payment_method, options)
+        return amount if amount.class == Response
+
         MultiResponse.run(:use_first_response) do |r|
-          r.process { authorize(100, credit_card, options) }
-          r.process(:ignore_result) { void(r.authorization, options) }
+          r.process { authorize(amount, payment_method, options) }
+          r.process(:ignore_result) { void(r.authorization, options) } unless amount == 0
         end
       end
 
+      def define_amount_for_verify(payment_method, options)
```

Comment:
Small thing, but `define` is redundant in ruby method names
```suggestion
      def amount_for_verify(payment_method, options)
```

Reply:
Thanks a lot! Solved!